modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/freelist/AbstractFreeList.java

```diff
@@ -751,6 +772,11 @@ public int emptyDataPages() {
         return initRecycledPage0(pageId, flag, initIO);
     }
 
+    /** {@inheritDoc} */
+    @Override protected int bucketsCnt() {
```

Comment:
We don't need this method for two reasons: first, it is used in `pageIds` method which I believe should be eliminated and second, there is already a field `buckets` in `PagesList` class with exactly this semantics.